### PR TITLE
Fixed typo, added tests and travis-ci config

### PR DESCRIPTION
**Pull request by kbond**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
+.idea
```
Comment (igorw) [inline .gitignore]: Please add this to your [global gitignore](https://help.github.com/articles/ignoring-files), there's no need to litter all projects with such specific rules.

Comment (igorw) [inline .gitignore]: I'll amend myself in this case.

Comment (kbond) [inline .gitignore]: ok, thanks for the tip

**Comment by kbond** [comment]
@igorw i amended the commit

**Comment by igorw** [comment]
I already had merged the previous commit, thanks for the fix and tests. I've added some more tests for the other response factories as well. Let's get this thing more break proof. ;-)

**Pull request closed without merging**